### PR TITLE
Cap concurrency when parsing blueprint files

### PR DESCRIPTION
**Pull request by colincross**
Darwin has a default limit of 256 open files per process.  Parsing
too many blueprint files in parallel can hit the limit.  Cap the
concurrency at 200.

Test: manual testing with limit set to 32
Change-Id: Ic64d21d2c0ffd7c86bf3f02fb51216ee5684a80c

### CODE REVIEW

File: context.go [hunk 1]
```diff
@@ -749,14 +755,19 @@ loop:
...
 			}
-
-			blueprintsSet[blueprint.string] = true
-			startParseBlueprintsFile(blueprint.string, blueprint.Scope)
+			startParseBlueprintsFile(blueprint)
 		case <-doneCh:
 			count--
+			if len(pending) > 0 {
+				startParseBlueprintsFile(pending[len(pending)-1])
+				pending = pending[0 : len(pending)-1]
```
Comment (Nan-Zhang) [inline context.go]: pending = pending[:len(pending) - 1]

File: context.go [hunk 2]
```diff
@@ -749,14 +755,19 @@ loop:
 			if tooManyErrors {
 				continue
 			}
-			if blueprintsSet[blueprint.string] {
+			// Limit concurrent calls to parseBlueprintFiles to 200
+			// Darwin has a default limit of 256 open files
```
Comment (danw) [inline context.go]: Any idea if this limits our performance? should we either auto-detect this, or have separate mac vs linux limits?

File: context.go (hunk 1 above)
Comment (colincross) [inline context.go]: Done

File: context.go (hunk 2 above)
Comment (colincross) [inline context.go]: I wouldn't think so - we want at least a few more goroutines than the number of cpus to allow for blocking on IO, but after that I'd expect adding more goroutines to reduce performance, not increase it.  200 is still probably more than we need.

**Pull request merged**